spring-cloud-task-core/src/main/java/org/springframework/cloud/task/configuration/observation/TaskDocumentedObservation.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.configuration.observation;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.observation.docs.DocumentedObservation;
+
+enum TaskDocumentedObservation implements DocumentedObservation {
+
+	/**
+	 * Observation created when a task runner is executed.
+	 */
+	TASK_RUNNER_OBSERVATION {
+		@Override
+		public String getName() {
+			return "spring.cloud.task.runner";
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return TaskRunnerTags.values();
+		}
+
+		@Override
+		public String getPrefix() {
+			return "spring.cloud.task";
```

Comment:
This update creates a breaking change where the original metric names are no longer valid.   For example the original query I played with looked something like:
`SELECT "upper" FROM "spring_cloud_task"  GROUP BY "spring_cloud_task_name"`
Now it looks like:
`SELECT "upper" FROM "spring_cloud_task_execution" GROUP BY "task_name"`
Do we want to  transform the metrics (enabled by property) using https://micrometer.io/docs/concepts#_transforming_metrics as an option so that their original queries will still work?   What are your thoughts on this?

Reply:
That's very good point...

Actually we could go the other way round. Leave what we have and by default migrate to legacy. That's because that way tracing tags will be correct and we would by default set metrics back to the legacy setup (this would be an opt-out feature, meaning you would have to explicitly state that you want to use the new tagging for metrics).

WDYT?

Reply:
That sounds great!

Reply:
Actually... The observation metrics are created as an addition to the ones coming from `TaskMetrics`. This PR doesn't remove any "legacy" metrics, so you should have both. Are you observing something different?

Reply:
Ok, that makes sense.    I will resolve the comment above then.